h/static/scripts/group-forms/hooks/use-group-annotations.tsx

```diff
@@ -10,36 +16,97 @@ export type GroupAnnotationsOptions = {
 };
 
 export type GroupAnnotationsResult = {
+  /** Whether there's data being currently loaded */
   loading: boolean;
+  /** Whether the first page of data is currently being loaded */
+  loadingFirstPage: boolean;
+  /** The last error that occurred, if any */
   error?: string;
+
+  /**
+   * The current list of annotations.
+   * It will be empty while loading the first page of data.
+   */
   annotations: APIAnnotationData[];
+
+  /**
+   * A callback to load the next up-to-20-item chunk of annotations, if any.
+   * Useful to progressively load more annotations, as this hook only loads the
+   * first chunk when invoked.
+   */
+  loadNextPage: () => void;
```

Comment:
This has been build for scroll-based pagination, and the page number is internally tracked. We could change this signature to allow passing the page number to load with not too many implementation changes, if we decided to finally add a regular pagination control to the UI.

Reply:
Has there been a discussion on what the final direction will be here?

Reply:
The decision was to go with scroll-based pagination for now, and only change it if we see it becomes a problem for users.